pyopenms_viz/_matplotlib/core.py

```diff
@@ -580,12 +580,93 @@ class MATPLOTLIBMobilogramPlot(MATPLOTLIBChromatogramPlot, MobilogramPlot):
     """
 
-    pass
+    def _create_figure(self):
+        if self.width is None or self.height is None:
+            self.fig, self.ax = plt.subplots()
+        else:
+            self.fig, self.ax = plt.subplots(figsize=(self.width / 100, self.height / 100), dpi=100)
 
+        self.ax.set_title(self.title)
+        self.ax.set_xlabel(self.xlabel)
+        self.ax.set_ylabel(self.ylabel)
 
+        # Ensure `self.canvas` references the Axes, not the Figure
+        self.canvas = self.ax
+        return self.ax
+
+    def create_main_plot(self):
+        """
+        Draw the main spectrum (e.g., peaks) on self.ax.
+        """
+        # Example: use a line plot for the spectrum
+        linePlot = self.get_line_renderer(
+            data=self.data,
+            x=self._config.x,
+            y=self._config.y,
+            config=self._config,
+            canvas=self.canvas,
+        )
+        ax = linePlot.generate(None, None)
+        return ax
+
+    def add_peptide_sequence(self, peptide_sequence: str, matched_fragments: List[Tuple[int, int]]):
+        """
+        Add a peptide sequence to the Matplotlib plot.
+
+        Parameters
+        ----------
+        peptide_sequence : str
+            The peptide sequence to be plotted.
+        matched_fragments : List[Tuple[int, int]]
+            List of tuples representing the start and end indices of matched fragments.
+        """
+        ax = self.canvas  # Get the current axes
+
+        # Get the y-axis limits to position the peptide sequence above the plot
+        y_min, y_max = ax.get_ylim()
+        sequence_y = y_max * 1.05  # Position the sequence slightly above the plot
+
+        # Plot the peptide sequence
+        for i, aa in enumerate(peptide_sequence):
+            ax.text(
+                i, sequence_y, aa, ha="center", va="bottom",
+                fontsize=self._config.peptide_sequence_fontsize,
+                color=self._config.peptide_sequence_color,
+            )
+
+        # Highlight matched fragments
+        for start, end in matched_fragments:
+            ax.axvspan(
+                start - 0.5, end + 0.5,
+                color=self._config.highlight_color,
+                alpha=self._config.highlight_alpha,
+            )
+
+        # Adjust the y-axis limits to accommodate the peptide sequence
+        ax.set_ylim(y_min, sequence_y * 1.1)
+
+    def plot(self):
+        """
+        Generate the base spectrum plot and add the peptide sequence if configured.
+        """
+
+        # Ensure we have a valid Axes
+        if self.canvas is None:
+            self.canvas = self._create_figure()
+
+        # Create the main spectrum plot
+        self.ax = self.create_main_plot()
+
+        # Plot peptide sequence if set in config
+        if self._config.display_peptide_sequence:
+            if "sequence" in self.data.columns:
+                self.add_peptide_sequence(self.data["sequence"], self.data["matched_fragments"])
+
+        return self.ax
```

Comment:
Would be best to add this instead to the _core.py SpectrumPlot's plot() function so that it can be shared across backends. I know only matplotlib is supported for now but it might be supported with other backends in the future.